cmd/tink-worker/container/container.go

```diff
@@ -0,0 +1,169 @@
+package container
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"reflect"
+	"regexp"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+)
+
+type cdata struct {
+	data configData
+}
+
+type configData struct {
+	name           string
+	pid            string
+	env            []string
+	volumeBindings []string
+	cmd            []string
+	imageName      string
+	registry       string
+	captureLogs    bool
+}
+
+type removeOpts struct {
+	force   bool
+	links   bool
+	volumes bool
+}
+
+// Create creates container and returns containerid on success else returns error.
+func (c cdata) Create(ctx context.Context, cli client.ContainerAPIClient) (string, error) {
+	fmt.Println(c.data)
+	if c.data.name == "" || c.data.imageName == "" {
+		return "", errors.New("container name and image name are required data")
+	}
+
+	if cli == nil || (reflect.ValueOf(cli).IsNil()) {
+		return "", errors.New("ContainerAPIClient interface is not valid")
+	}
+
+	if len(c.data.cmd) == 0 {
+		return "", errors.New("cmd has invalid data")
+	}
+
+	if len(c.data.volumeBindings) > 0 {
+		isValidVolume := regexp.MustCompile(`^[a-zA-Z0-9/:-]*$`).MatchString
+		for _, value := range c.data.volumeBindings {
+			if !isValidVolume(value) {
+				return "", errors.New("volume has invalid data")
+			}
+		}
+	}
+
+	config := &container.Config{
+		Image:        path.Join(c.data.registry, c.data.imageName),
+		AttachStdout: true,
+		AttachStderr: true,
+		Cmd:          c.data.cmd,
+		Tty:          true,
+		Env:          c.data.env,
+	}
+	if !c.data.captureLogs {
+		config.AttachStdout = false
+		config.AttachStderr = false
+		config.Tty = false
+	}
+
+	hostConfig := &container.HostConfig{
+		Privileged: true,
+		Binds:      c.data.volumeBindings,
+	}
+
+	if pidConfig := c.data.pid; pidConfig != "" {
+		hostConfig.PidMode = container.PidMode(pidConfig)
+	}
+
+	resp, err := cli.ContainerCreate(ctx, config, hostConfig, nil, nil, c.data.name)
+	if err != nil {
+		return "", errors.Wrap(err, "DOCKER CREATE")
+	}
+	return resp.ID, nil
+}
+
+// Start starts container and returns error on failure.
+func (c cdata) Start(ctx context.Context, cli client.ContainerAPIClient, id string) error {
+	if id == "" {
+		return errors.New("container id is not valid")
+	}
+
+	if cli == nil || (reflect.ValueOf(cli).IsNil()) {
+		return errors.New("ContainerAPIClient interface is not valid")
+	}
+
+	err := cli.ContainerStart(ctx, id, types.ContainerStartOptions{})
+	if err != nil {
+		return errors.Wrap(err, "DOCKER START")
```

Comment:
does `DOCKER START` add value or help the user of this function understand the problem better? just-food for thought.